pkg/sendconfig/sendconfig.go

```diff
@@ -138,7 +138,7 @@ func onUpdateInMemoryMode(ctx context.Context,
 
 	_, err = kongConfig.Client.Do(ctx, req, nil)
 	if err != nil {
-		return fmt.Errorf("posting new config to /config: %w", err)
+		return fmt.Errorf("posting new config (%s) to /config: %w", string(config), err)
```

Comment:
I added this because it made figuring out what went wrong from the controller logs a LOT easier for me when an error occurred.

Reply:
https://github.com/Kong/kubernetes-ingress-controller/pull/991 should handle this and is in current next. It may have still been open when work on this started originally. Filesystem usage is a bit clunky (we may want to figure out a way to expose the good/bad configs on the metrics server in the future), but intentional to avoid leaking certificate keys and the like in logs. Config blobs can also get quite large in practice also, so the single set of files is also an anti-logspam measure.

Reply:
Sounds good, changing this back for now :+1: 

```suggestion
		return fmt.Errorf("posting new config to /config: %w", err)
```